wavefront_opentracing_python_sdk/tracer.py

```diff
@@ -0,0 +1,194 @@
+        :param tags: Tags
+        :type tags: list of pair
+        """
+        super(WavefrontTracer, self).__init__(ThreadLocalScopeManager())
+        self._reporter = reporter
+        self._tags = tags or []
+        self.registry = PropagatorRegistry()
+        # self.random = random.Random(time.time() * (os.getpid() or 1))
+
+    # pylint: disable=too-many-arguments,too-many-locals
+    def start_span(self,
+                   operation_name=None,
+                   child_of=None,
+                   references=None,
+                   tags=None,
+                   start_time=None,
+                   ignore_active_span=False):
+        r"""
+        Start and return a new :class:`Span` representing a unit of work.
+
+        :param operation_name: Operation Name
+        :type operation_name: str
+        :param child_of: (optional) a WavefrontSpan or WavefrontSpanContext
+            instance representing the parent in a REFERENCE_CHILD_OF reference.
+            If specified, the `references` parameter must be omitted.
+        :type child_of: WavefrontSpanContext or WavefrontSpan
+        :param references: (optional) references that identify one or more
+            parent :class:`SpanContext`\ s. (See the Reference documentation
+            for detail).
+        :type references: `list` of `opentracing.Reference`
+        :param tags: an optional list of tags.
+        :type tags: list of pair
+        :param start_time: an explicit Span start time as a unix timestamp per
+            :meth:`time.time()`
+        :type start_time: float
+        :param ignore_active_span: an explicit flag that ignores the current
+            active :class:`Scope` and creates a root :class:`Span`.
+        :type ignore_active_span: bool
+        :return: an already-started :class:`WavefrontSpan` instance.
+        :rtype: WavefrontSpan
+        """
+        parents = []
+        follows = []
+        baggage = None
+        tags = tags or []
+        tags.extend(self._tags)
+        start_time = start_time or time.time()
+
+        parent = child_of
+        if parent is None and references:
+            one_reference = references
+            if isinstance(references, list):
+                one_reference = references[0]
+                for reference in references:
+                    if reference.get_type() == ReferenceType.CHILD_OF:
+                        parents.append(
+                            reference.referenced_context.get_span_id())
+                    elif reference.get_type() == ReferenceType.FOLLOWS_FROM:
+                        follows.append(
+                            reference.referenced_context.get_span_id())
+            parent = one_reference.referenced_context
+        # allow Span to be passed as reference, not just SpanContext
+        if isinstance(parent, WavefrontSpan):
+            parent = parent.context
+
+        if ignore_active_span or parent is None or not parent.has_trace:
```

Comment:
This is incorrect. If the `parent is not None` we should always use that for the traceId (even if `ingore_active_span` is true).

Also if `parent is None` and `ignore_active_span` is False we should use the current active span obtained from the tracer/scope manager to set the parent/traceId.